qiskit_nature/second_q/hamiltonians/lattices/kagome_lattice.py

```diff
@@ -0,0 +1,417 @@
+            # The periodic boundary condition in the x direction.
+            # It makes sense only when rows is greater than 1.
+            if rows > 1:
+                for y in range(cols):
+                    cell_a_idx = self._coordinate_to_index(np.array([rows - 1, y]))
+                    cell_a_1 = num_sites_per_cell * cell_a_idx + 1
+
+                    cell_b_idx = self._coordinate_to_index(np.array([0, y]))
+                    cell_b_0 = num_sites_per_cell * cell_b_idx
+
+                    list_of_edges.append((cell_a_1, cell_b_0, edge_parameter.conjugate()))
+            # The periodic boundary condition in the y direction.
+            # It makes sense only when cols is greater than 1.
+            if cols > 1:
+                for x in range(rows):
+                    cell_a_idx = self._coordinate_to_index(np.array([x, cols - 1]))
+                    cell_a_2 = num_sites_per_cell * cell_a_idx + 2
+
+                    cell_b_idx = self._coordinate_to_index(np.array([x, 0]))
+                    cell_b_0 = num_sites_per_cell * cell_b_idx
+
+                    list_of_edges.append((cell_a_2, cell_b_0, edge_parameter.conjugate()))
+
+            # The periodic boundary condition in the diagonal directions.
+            for x in range(1, rows):
+                cell_a_idx = self._coordinate_to_index(np.array([x, cols - 1]))
+                cell_a_2 = num_sites_per_cell * cell_a_idx + 2
+
+                cell_b_idx = self._coordinate_to_index(np.array([(x - 1) % rows, 0]))
+                cell_b_1 = num_sites_per_cell * cell_b_idx + 1
+
+                list_of_edges.append((cell_a_2, cell_b_1, edge_parameter.conjugate()))
+
+            for y in range(cols - 1):
+                cell_a_idx = self._coordinate_to_index(np.array([0, y]))
+                cell_a_2 = num_sites_per_cell * cell_a_idx + 2
+
+                cell_b_idx = self._coordinate_to_index(np.array([rows - 1, (y + 1) % cols]))
+                cell_b_1 = num_sites_per_cell * cell_b_idx + 1
+
+                list_of_edges.append((cell_a_2, cell_b_1, edge_parameter.conjugate()))
+
+            # isolating x = 0, y = cols - 1 to prevent duplicating edges
+            cell_a_idx = self._coordinate_to_index(np.array([0, cols - 1]))
+            cell_a_2 = num_sites_per_cell * cell_a_idx + 2
+
+            cell_b_idx = self._coordinate_to_index(np.array([rows - 1, 0]))
+            cell_b_1 = num_sites_per_cell * cell_b_idx + 1
+
+            list_of_edges.append((cell_a_2, cell_b_1, edge_parameter.conjugate()))
+
+        elif boundary_condition == BoundaryCondition.OPEN:
+            pass
+        else:
+            raise ValueError(
+                f"Invalid `boundary condition` {boundary_condition} is given."
+                "`boundary condition` must be " + " or ".join(str(bc) for bc in BoundaryCondition)
+            )
+        return list_of_edges
+
+    def _default_position(self, with_boundaries: bool = True) -> dict[int, list[float]]:
```

Comment:
```suggestion
    def _default_position(self, *, with_boundaries: bool = True) -> dict[int, list[float]]:
```
This makes this a keyword-only argument which I think is appropriate here